### DISCUSSION TITLE
GPU Power limits for 2024 models

### DISCUSSION THREAD

**Discussion by seerge** [seerge/g-helper#2141] [Announcements]
Hello,

Recently I have discovered that in 2024 models next to already existing GPU ``Dynamic Boost`` (ranging from 5W to 25W usually) new devices also have adjustable ``GPU power`` slider (!)

It's a "base" power, so your total GPU power limit will be ``GPU Power`` + ``Dynamic Boost`` (as it always have been)

<table><tr>
<td width="50%"><img src="https://github.com/seerge/g-helper/assets/5920850/28e71141-50ce-4b30-a8d0-a783300aa72b"></td>
<td width="50%"><img src="https://github.com/seerge/g-helper/assets/5920850/de5d08bb-b4b9-45ac-aedf-4a1ef9f2323c"></td>
</tr></table>

Please keep in mind, that next to this setting, your GPU is also limited internally by it's own VBIOS (Video BIOS). And if your GPU has for exaple 125W VBIOS, it won't go above that no matter what :) 

Best way to test limit  is to download latest version of [FURMARK](https://geeks3d.com/furmark/downloads/) and [HWinfo](https://www.hwinfo.com/). Run benchmark in a window and just monitor GPU power draw via HWinfo

If you are providing feedback, or believe something is not working as it should, please either post a [new issue](https://github.com/seerge/g-helper/issues/new/choose) or include full app logs and clear scenario to reproduce as a comment here

### [Changes are live in latest releases](https://github.com/seerge/g-helper/releases)

**Comment by xSimone** [comment]
💙💙💙

**Comment by famhu** [comment]
Is this not suitable for the 2023 model?

**Reply by seerge** [reply]
@famhu it's based on a new functionality asus has added in 2024 devices :) you can try this build - if you won't see a power slider - it means your device doesn't support it

**Reply by famhu** [reply]
I tried it, but I can't control the power consumption of the GPU on my computer. My model is G834JZ

**Reply by xSimone** [reply]
G834JZ is 2023 model not 2024 :x

**Reply by redlock81** [reply]
it does not work on my Strix G18 (2024) 14900hx 4080 it stays at 150w, AC keeps it boosted to 175w dynamic boost needs work, of course i thank you for your efforts!!!!!

**Reply by seerge** [reply]
Base power will be in range up to 150W,  and with boost (+25W) it's total 175W. You can check that in Furmark.

This endpoints are SAME for all devices. Even without setting any GPU power limits, it will be maximized under Turbo mode.

P.S. This comment provides 0 helpful information. If you believe something is not working as it should, you need to provide a clear scenario to reproduce and upload logs as explained in [issue template](https://github.com/seerge/g-helper/issues/new/choose)

**Comment by ModrnJosh** [comment]
Thank you Seerge, this is incredible!!! This essentially gives us functionality similar to the TGP slider in MSI Afterburner. Tested on the 4090 G16 and I can now achieve the full 125W even in Turbo or Balanced modes instead of being restricted to Manual mode for full TGP, wow!!

Also, this could be useful for vbios mods, which I will also test. This could mean potentially that I could load up a 175W vbios on the 2024 G16 and run it at maybe 150W instead to keep things cooler without having to limit clock speeds or anything. That would put the 2024 G16 at essentially the same power level as the 2023 Zephyrus M16, which is thicker and heavier.

**Reply by seerge** [reply]
@ModrnJosh great news. So everything is working as it should in this build ? In what range can you set GPU power and what happens if you go above (i assume you are locked with 125W) ?

Does it work for G14 ?

**Reply by ModrnJosh** [reply]
Hey, I know we’ve chatted about this a bit now but for anyone else reading, going above 125W doesn’t seem to boost the wattage any higher. It may or may not change the way the CPU acts towards dynamic boost, but I’ll be playing with it more. For now in the limited testing I’ve done, there doesn’t seem to be a difference between 105W + 20W dynamic boost and 125W + 25W dynamic boost as the machine maxes out at 125W regardless. And then I know you’ve tested it now a bit yourself on the G14, but I haven’t had much time to yet.

**Comment by seerge** [comment]
Changes are live in latest release https://github.com/seerge/g-helper/releases

**Reply by xSimone** [reply]
Yay :3

**Reply by seerge** [reply]
Thanks for testing :)

**Comment by SylvesZhou** [comment]
4060 G14 2024 seem not have the GPU Power limits function.

**Reply by seerge** [reply]
It does:) as I have G14 2024. Do you use latest release?

**Comment by alex450** [comment]
maybe 4070 laptop got a wrong GPU POWER?（should be 115+25=140W）
tested at TX5pro (tuf gaming a16 chinese special version)
asus offically claims 140w for its 4070,and when using previous build and went furmark,it could reach more than 130watts
when v0.158, its locked at 110watts

![QQ截图20240309062927](https://github.com/seerge/g-helper/assets/17879601/a6adcb77-8de0-41c7-b7e3-eb871f24b14b)

**Reply by seerge** [reply]
That's the first time I hear that such thing even exists, what's your full model name ? Can you run a scan from this thread https://github.com/seerge/g-helper/issues/38 and post results here ?

**Reply by seerge** [reply]
GPU Power on Asus devices consists of 2 parts : "base" value (reported by your bios) and "variable" part that can be set.

By default "variable" part could go in range from 0W to 50W. Since you have 105W on your screenshot, I assume that "base" was 55W correspondingly (so you can't limit lower than 55W)

I have increased "variable" range to 60W by default in this build

[GHelper.zip](https://github.com/seerge/g-helper/files/14543887/GHelper.zip)

P.S. Did you have GPU power slider in AC ?

**Reply by alex450** [reply]
[myacresult.zip](https://github.com/seerge/g-helper/files/14544514/myacresult.zip)
method 2
model name in chinese（I beleive they only provide this in china）：天选5pro锐龙版
you may find something here:

https://www.techpowerup.com/318269/amd-ryzen-9-7940hx-apu-benchmarked-in-asus-tianxuan-5-pro-laptop
https://www.youtube.com/watch?v=wjV2dZrnJdI

**Reply by alex450** [reply]
![IMG_20240309_093500](https://github.com/seerge/g-helper/assets/17879601/c3beba10-c9c4-4da1-a097-39abfdeee9c2)
so the default range is 55 to 115

**Reply by alex450** [reply]
![图片](https://github.com/seerge/g-helper/assets/17879601/5d3a76a3-f77e-4645-898d-8701019e7e38)
vbios support 140W
![图片](https://github.com/seerge/g-helper/assets/17879601/e92fd515-a4b3-4c34-a310-1895d344d6f7)

**Reply by ModrnJosh** [reply]
Wow! That's a unique laptop. 7940HX with 140W 4070, interesting. Would you mind extracting the vbios from GPU-Z? The little "share" icon next to the "uefi" checkbox. I'd love to test this on the G14. That's a crazy wattage range too. 55 to 115.

**Reply by xSimone** [reply]
Yes. This is a very weird Laptop. :<

**Reply by seerge** [reply]
@alex450 great, does the build I linked work as intended?

**Reply by seerge** [reply]
@alex450 @ModrnJosh 

I have changed logic of Max GPU Power setting a bit. So instead of just hardcoding it for each model, app would simply read it from ``nvidia-smi --query-gpu=power.max_limit --format csv,noheader,nounits`` command. (i.e. show "actual" power limit , based on what nvidia reports)

This should also accommodate situations when people use custom VBIOS (as @ModrnJosh does)

[GHelper.zip](https://github.com/seerge/g-helper/files/14546281/GHelper.zip)

**Reply by alex450** [reply]
after solving this issue i would upload it

**Reply by seerge** [reply]
@alex450 what ?

**Comment by alex450** [comment]
> @alex450 @ModrnJosh
> 
> I have changed logic of Max GPU Power setting a bit. So instead of just hardcoding it for each model, app would simply read it from `nvidia-smi --query-gpu=power.max_limit --format csv,noheader,nounits` command. (i.e. show "actual" power limit , based on what nvidia reports)
> 
> This should also accommodate situations when people use custom VBIOS (as @ModrnJosh does)
> 
> [GHelper.zip](https://github.com/seerge/g-helper/files/14546281/GHelper.zip)

this build works even better than former version! 
when going furmark,it went straightly 140watts instead of floating between 120w and 140w

**Reply by seerge** [reply]
Great news, thanks for testing

**Reply by alex450** [reply]
![图片](https://github.com/seerge/g-helper/assets/17879601/4afd6acd-8c29-4679-b5cb-f3cc0d511f15)
stably running 
settings
![图片](https://github.com/seerge/g-helper/assets/17879601/3803145b-9706-474c-8c46-27a711ded157)

@ModrnJosh vbios here https://github.com/alex450/140W-4070-laptop-vbios/tree/main

**Reply by ModrnJosh** [reply]
Thank you!!!

**Comment by l0pl0p** [comment]
This is a great feature!  I just grabbed a G14 today and am testing it--it isn't working for me. With GHelper installed and all ASUS services stopped/disabled/no armory crate, I am locked at 55W actual GPU no matter what I set it at in GHelper.  

With armory crate turbo, I get ~90W.

Let me know if there's anything you'd like me to test!

**Reply by seerge** [reply]
@l0pl0p it's working absolutely fine, and tested very thoughtfully on G14. There is no "G-helper's" Turbo, all modes are baked in BIOS and are absolutely same.

You have apparently dragged power slider to 55w or did something wrong to your device.

If you want to apply custom limits, don't forget to check ``Apply``. If you want to go back to all defaults how it was set by Asus - click ``Factory Defaults``

**Reply by seerge** [reply]
@l0pl0p if you are interested, you may do same as another user in this comment https://github.com/seerge/g-helper/discussions/2141#discussioncomment-8893180 (reinstall latest nvidia driver), as this is likely driver issue rather g-helper's

**Comment by vansonffff** [comment]
when the G14(2024) resume from sleep，the GPU will be limited on 55W. 
changing the batterymode（eco，balance，power) is not helpful，unless restart the laptop.

**Reply by seerge** [reply]
@vansonffff how do you test actual power limit? Please post screenshots of a measurement and logs as was asked. Thanks

**Reply by vansonffff** [reply]
![屏幕截图 2024-03-22 133600](https://github.com/seerge/g-helper/assets/162583749/234bb6a2-da81-449f-9158-b7277a9af018)
![屏幕截图 2024-03-22 133408](https://github.com/seerge/g-helper/assets/162583749/c0aa10d4-5f59-4615-a5f7-4dea80b841d0)

gpu limited on 55w

**Reply by seerge** [reply]
@vansonffff can you post a log as I was asking? Thanks

`%APPDATA%\GHelper\log.txt`

Also I see that GPU power slider shows 80w, while it can’t reach 65w on G14 2024. Do you use custom VBIOS or something like that?

**Reply by vansonffff** [reply]
NO，the bios is original.
I assume  that could be a software conflict between ARMOURY CRATE）？

**Reply by seerge** [reply]
@vansonffff if you still have AC it can interfere of course

**Reply by vansonffff** [reply]
sorry，i can‘t get your point.
i use original ASUS AC adapter while using the G14(2024)

**Reply by vansonffff** [reply]
or it might a BUG on bios?
someone reflects same thing on G14(2023) about GPU55W

**Comment by vansonffff** [comment]
I‘ve uninstall ARMOURY CRATE，but GPU still limited on 55w

**Reply by seerge** [reply]
@vansonffff you have been explicitly asked to provide logs twice. Since you ignore this request all the time, it's impossible to assist you further. Thanks.

I have the same device as you claim to have (G14 2024) and can not reproduce behavior you indicate.

**Reply by vansonffff** [reply]
I'm sorry, I check the nvidia-msi ，I hope the new screenshoots may help，really appreciate .
![屏幕截图 2024-03-23 195909](https://github.com/seerge/g-helper/assets/162583749/b1259649-773d-4ae7-9d1b-ae01ba207937)
![屏幕截图 2024-03-23 200328](https://github.com/seerge/g-helper/assets/162583749/15ab8953-9bb9-4414-820d-2d0460f5aa74)
![屏幕截图 2024-03-23 200346](https://github.com/seerge/g-helper/assets/162583749/46ba220a-b426-4c7a-9522-c944f43b22d5)

**Reply by seerge** [reply]
@vansonffff G-Helper **is not setting any limits on a driver level**, and your screenshots show that nvidia-smi see a driver level limitation. It's some other software or Nvidia setting (like whisper mode or so that does it for you.

**Reply by vansonffff** [reply]
Thank you！
Follow your advice，I install the old versioin of Nvidia drive，and uninstall the NV EXP，it seems the problem has been solved.
and GPU power limits on the latest version of ghelper is working.

**Reply by seerge** [reply]
@vansonffff ok, thanks for confirming that it was Nvidia driver issue.
